flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/factories/TestTimeTravelCatalog.java

```diff
@@ -0,0 +1,102 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.factories;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.table.api.Schema;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.catalog.GenericInMemoryCatalog;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.catalog.exceptions.DatabaseNotExistException;
+import org.apache.flink.table.catalog.exceptions.TableAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/** Test Catalog for testing time travel. */
+public class TestTimeTravelCatalog extends GenericInMemoryCatalog {
+    public TestTimeTravelCatalog(String name) {
+        super(name);
+
+        this.tables = new HashMap<>();
+    }
+
+    private final Map<ObjectPath, List<Tuple2<Long, CatalogTable>>> tables;
+
+    @Override
+    public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ignoreIfExists)
+            throws TableAlreadyExistException, DatabaseNotExistException {
+        super.createTable(tablePath, table, ignoreIfExists);
+    }
+
+    @Override
+    public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistException {
+        return super.getTable(tablePath);
+    }
+
+    @Override
+    public CatalogBaseTable getTable(ObjectPath tablePath, long timestamp)
+            throws TableNotExistException {
+
+        if (tables.containsKey(tablePath)) {
+            List<Tuple2<Long, CatalogTable>> tableList = tables.get(tablePath);
+
+            Optional<Tuple2<Long, CatalogTable>> table =
+                    tableList.stream()
+                            .filter(t -> t.f0 <= timestamp)
+                            .max(Comparator.comparing(t -> t.f0));
+
+            if (table.isPresent()) {
+                return table.get().f1;
+            }
+        }
+
+        return super.getTable(tablePath);
+    }
+
+    /**
+     * @param tableName Table name
+     * @param schema Table schema of the table
+     * @param properties Table properties to construct a table instance
+     * @param timestamp The snapshot of the table
+     */
+    public void registerTable(
```

Comment:
```suggestion
    public void registerTableForTableTravel(
```
?
Also, could you please add a comment for this method? I did spend somtime to under the logic that 
it'll first add to `tables` and then add it again to `tables` in `GenericInMemoryCatalog`?